sdk/src/main/java/io/dapr/config/Property.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ */
+
+package io.dapr.config;
+
+/**
+ * A configuration property in the Dapr's SDK.
+ */
+public abstract class Property<T> {
+
+  /**
+   * Property's name as a Java Property.
+   */
+  private final String name;
+
+  /**
+   * Property's name as a environment variable.
+   */
+  private final String envName;
+
+  /**
+   * Default value.
+   */
+  private final T defaultValue;
+
+  /**
+   * Instantiates a new configuration property.
+   *
+   * @param name Java property name.
+   * @param envName Environment variable name.
+   * @param defaultValue Default value.
+   */
+  Property(String name, String envName, T defaultValue) {
+    this.name = name;
+    this.envName = envName;
+    this.defaultValue = defaultValue;
+  }
+
+  /**
+   * Gets the Java property's name.
+   * @return Name.
+   */
+  public String getName() {
+    return this.name;
+  }
+
+  /**
+   * Gets the environment variable's name.
+   * @return Name.
+   */
+  public String getEnvName() {
+    return this.envName;
+  }
+
+  /**
+   * Gets the value defined by system property first, then env variable or sticks to default.
+   *
+   * @return Value from system property (1st) or env variable (2nd) or default (last).
+   */
+  public T get() {
+    String propValue = System.getProperty(this.name);
+    if (propValue != null && !propValue.trim().isEmpty()) {
+      try {
+        return this.parse(propValue);
+      } catch (Exception e) {
+        e.printStackTrace();
+        // OK, we tried. Falling back to system environment variable.
+      }
+    }
+
+    String envValue = System.getenv(this.envName);
+    if (envValue != null && !envValue.trim().isEmpty()) {
+      try {
+        return this.parse(envValue);
+      } catch (Exception e) {
+        e.printStackTrace();

```

Comment:
Same here.

Reply:
We do not log anything in the SDK, we throw exceptions and expect user code to handle those. I will just silence the exception here.